### PR TITLE
V abt/tab device permissions

### PR DESCRIPTION
**Pull request by AbhijeetThakur-msft**
Added the following capabilities in the sample:

- Request device permissions for your Microsoft Teams app
- Integrate media capabilities
- Integrate QR or barcode scanner capability
- Integrate location capabilities
- Integrate People Picker capability

### CODE REVIEW

File: samples/tab-device-permissions/nodejs/appPackage/manifest.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    "icons": {
+        "color": "color.png",
+        "outline": "outline.png"
+    },
+    "name": {
+        "short": "mgt-teams-tab",
+        "full": ""
+    },
+    "description": {
+        "short": "Short description for Personal App.",
```
Comment (v-chetsh) [inline samples/tab-device-permissions/nodejs/appPackage/manifest.json:22]: Can we add App summary here?

File: samples/tab-device-permissions/nodejs/appPackage/manifest.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        "websiteUrl": "https://50c2552618db.ngrok.io",
+        "privacyUrl": "https://50c2552618db.ngrok.io/privacy",
+        "termsOfUseUrl": "https://50c2552618db.ngrok.io/termsofuse"
+    },
+    "icons": {
+        "color": "color.png",
+        "outline": "outline.png"
+    },
+    "name": {
+        "short": "mgt-teams-tab",
```
Comment (v-chetsh) [inline samples/tab-device-permissions/nodejs/appPackage/manifest.json:18]: Please provide proper app name

File: samples/tab-device-permissions/nodejs/appPackage/manifest.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
+{
+    "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.8/MicrosoftTeams.schema.json",
+    "manifestVersion": "1.8",
+    "version": "1.0.1",
```
Comment (v-chetsh) [inline samples/tab-device-permissions/nodejs/appPackage/manifest.json:4]: I think version should be 1.0.0

File: samples/tab-device-permissions/nodejs/appPackage/manifest.json [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    ],
+    "permissions": [
+        "identity",
+        "messageTeamMembers"
+    ],
+    "validDomains": [
+        "*.ngrok.io"
+    ],
+    "devicePermissions": [
+        "media"
```
Comment (v-chetsh) [inline samples/tab-device-permissions/nodejs/appPackage/manifest.json]: What about the device permissions for location

File: samples/tab-device-permissions/nodejs/src/components/CaptureImage.tsx [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+            } else {
+              alert(" ErrorCode: " + error.errorCode);
+            }
+          } else if (files) {
+            setCapturedImage(files[0].content);
+          }
+    });
+ }
+ 
+  //method to validate before capturing media
```
Comment (v-chetsh) [inline samples/tab-device-permissions/nodejs/src/components/CaptureImage.tsx:38]: Add space & capital M

File: samples/tab-device-permissions/nodejs/src/components/CaptureImage.tsx [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+        </CardHeader>
+        <CardBody>
+        <Flex column gap="gap.small">
+        <Text content="Checks for permission before capturing image" />
+          <Button content="Capture Image" onClick={captureImage}/>
+        </Flex>
+        </CardBody>
+        {capturedImage != '' && 
+        <Layout styles={{ maxWidth: '150px',}}
+        debug
```
Comment (v-chetsh) [inline samples/tab-device-permissions/nodejs/src/components/CaptureImage.tsx:90]: Remove debugger

File: samples/tab-device-permissions/nodejs/src/components/CaptureImage.tsx [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+        <Carousel
+         ariaRoleDescription="carousel"
+         ariaLabel="Selected images"
+         navigation={{
+            'aria-label': 'selected images',
+             items: capturedImages.map((item, index) => ({
+             key: index,
+             'aria-controls': item.id,
+              content: item.thumbnail,
+           })),
```
Comment (v-chetsh) [inline samples/tab-device-permissions/nodejs/src/components/CaptureImage.tsx:119]: Is it aligned properly, Please check

File: samples/tab-device-permissions/nodejs/src/components/GetGeoLocation.tsx [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+        </CardHeader>
+        <CardBody>
+        <Flex column gap="gap.small">
+          <Button content="Get Location" onClick={getLocation}/>
+         {geoLocationValue != '' && 
+         <Text content={geoLocationValue}></Text>}
+          <Button content="Show Location" onClick={showLocation}/>
+        </Flex>
+        </CardBody>
+
```
Comment (v-chetsh) [inline samples/tab-device-permissions/nodejs/src/components/GetGeoLocation.tsx:62]: Remove empty extra lines

File: samples/tab-device-permissions/nodejs/src/components/PeoplePicker.tsx [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+ * The 'PeoplePicker' component for 
+ * of your app.
+ */
+const PeoplePicker = () => {
+ const [selectedPeople, setSelectedPeople] = useState<microsoftTeams.people.PeoplePickerResult[]>([]);
+ useEffect(() => {
+  microsoftTeams.initialize()
+ })
+ 
+ // method to select people using people picker control
```
Comment (v-chetsh) [inline samples/tab-device-permissions/nodejs/src/components/PeoplePicker.tsx:18]: Capital M

File: samples/tab-device-permissions/nodejs/src/components/PeoplePicker.tsx [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+           }
+            else 
+            {
+              alert(" ErrorCode: " + error.errorCode);
+            }
+      }
+    if (people)
+     {
+            setSelectedPeople(people);
+      }
```
Comment (v-chetsh) [inline samples/tab-device-permissions/nodejs/src/components/PeoplePicker.tsx:37]: Please check if contents are aligned properly

File: samples/tab-device-permissions/nodejs/appPackage/manifest.json (hunk 1 above)
Comment (v-kalc) [reply samples/tab-device-permissions/nodejs/appPackage/manifest.json]: Updated

File: samples/tab-device-permissions/nodejs/appPackage/manifest.json (hunk 2 above)
Comment (v-kalc) [reply samples/tab-device-permissions/nodejs/appPackage/manifest.json]: Updated App name

File: samples/tab-device-permissions/nodejs/src/components/CaptureImage.tsx (hunk 7 above)
Comment (v-kalc) [reply samples/tab-device-permissions/nodejs/src/components/CaptureImage.tsx]: Fixed alignment issues

File: samples/tab-device-permissions/nodejs/src/components/CaptureImage.tsx (hunk 6 above)
Comment (v-kalc) [reply samples/tab-device-permissions/nodejs/src/components/CaptureImage.tsx]: Removed

File: samples/tab-device-permissions/nodejs/src/components/GetGeoLocation.tsx (hunk 8 above)
Comment (v-kalc) [reply samples/tab-device-permissions/nodejs/src/components/GetGeoLocation.tsx]: Removed extra line

File: samples/tab-device-permissions/nodejs/src/components/PeoplePicker.tsx (hunk 9 above)
Comment (v-kalc) [reply samples/tab-device-permissions/nodejs/src/components/PeoplePicker.tsx]: Made M capital

File: samples/tab-device-permissions/nodejs/src/components/PeoplePicker.tsx (hunk 10 above)
Comment (v-kalc) [reply samples/tab-device-permissions/nodejs/src/components/PeoplePicker.tsx:30]: Fixed alignment issues

File: samples/tab-device-permissions/nodejs/appPackage/manifest.json (hunk 2 above)
Comment (v-kalc) [reply samples/tab-device-permissions/nodejs/appPackage/manifest.json]: Updated App name

File: samples/tab-device-permissions/nodejs/appPackage/manifest.json (hunk 3 above)
Comment (v-kalc) [reply samples/tab-device-permissions/nodejs/appPackage/manifest.json]: Updated to 1.0.0

File: samples/tab-device-permissions/nodejs/src/components/CaptureImage.tsx (hunk 5 above)
Comment (v-kalc) [reply samples/tab-device-permissions/nodejs/src/components/CaptureImage.tsx]: Made M capital

**Review by v-chetsh — APPROVED** [review]
Changes looks good.

File: samples/outgoing-webhook/nodejs/README.md [hunk 11]
```diff
@@ -1,56 +1,43 @@
----
```
Comment (Wajeed-msft) [inline samples/outgoing-webhook/nodejs/README.md:1]: Could you please revert this change? Seems like change from other sample.

Comment (Wajeed-msft) [reply samples/outgoing-webhook/nodejs/README.md:1]: I see multiple files from samples/outgoing-webhook/nodejs/ folder. Please revert them all.

File: samples/tab-device-permissions/nodejs/README.md [hunk 12]
```diff
@@ -0,0 +1,38 @@
+# Tab Device Permission Demo
+This is demo app to for tab device permissions.
+## Getting started
+1. Install some sort of tunnelling service. These instructions assume you are using ngrok: https://ngrok.com/
+1. Begin your tunnelling service to get an https endpoint. For this example ngrok is used. Start an ngrok tunnel with the following command (you'll need the https endpoint for the bot registration):<br>
+    ```bash
+    ngrok http 3978 --host-header=localhost
```
Comment (Wajeed-msft) [inline samples/tab-device-permissions/nodejs/README.md:7]: This sample runs by default on Port no. 3000. Could you please update this accordingly?

File: samples/tab-device-permissions/nodejs/README.md [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+    ```
+1. Create an app manifest. Navigate to the file, manifest/manifest.json - Change:
+    1. <<BASE_URI_DOMAIN>> change to your https endpoint from ngrok excluding the "https://" part
+    1. Save the file and zip this file along with both the png icons (located next to it) together to create a manifest.zip file
+## See it running
+1. Clone the sample and open it in Visual Studio. (Look at [similar steps if you are not sure](https://docs.microsoft.com/microsoftteams/platform/tutorials/get-started-dotnet-app-studio#download-the-sample))
+1. [Build and run your sample](https://docs.microsoft.com/microsoftteams/platform/tutorials/get-started-dotnet-app-studio#build-and-run-the-sample).<br/>
```
Comment (Wajeed-msft) [inline samples/tab-device-permissions/nodejs/README.md:14]: This is currnely navigating to Build and run the sample C# .NET. Can we include the `npm install` & `npm start` directly in this readme instead of redirecting?

File: samples/tab-device-permissions/nodejs/appPackage/README.md [hunk 14]
```diff
@@ -0,0 +1,8 @@
+# App Package
```
Comment (Wajeed-msft) [inline samples/tab-device-permissions/nodejs/appPackage/README.md:1]: Can we remove this ReadMe.md file as this doesn't add much value?

File: samples/tab-device-permissions/nodejs/appPackage/manifest.json [hunk 15]
```diff
@@ -0,0 +1,48 @@
+{
+    "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.8/MicrosoftTeams.schema.json",
+    "manifestVersion": "1.8",
+    "version": "1.0.0",
+    "id": "15a789f3-c143-43c8-bf1c-fe845718fe70",
+    "packageName": "com.microsoft.teams.extension",
+  "developer": {
+    "name": "Teams App, Inc.",
```
Comment (Wajeed-msft) [inline samples/tab-device-permissions/nodejs/appPackage/manifest.json:8]: Could you please correct the indentation of this file?

File: samples/tab-device-permissions/nodejs/src/components/PeoplePicker.tsx [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+ * of your app.
+ */
+const PeoplePicker = () => {
+ const [selectedPeople, setSelectedPeople] = useState<microsoftTeams.people.PeoplePickerResult[]>([]);
+ useEffect(() => {
+  microsoftTeams.initialize()
+ })
+ 
+ // Method to select people using people picker control
+ function selectPeople()
```
Comment (Wajeed-msft) [inline samples/tab-device-permissions/nodejs/src/components/PeoplePicker.tsx:19]: Could you please format all these code files? 
image

File: samples/tab-device-permissions/nodejs/src/components/Tab.tsx [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+import CaptureAudio from './CaptureAudio';
+
+/**
+ * The 'Tab' contains all the components
+ * of your app.
+ */
+const Tab = () => {
+    return(
+      <Flex column gap="gap.large">
+       {/* Component to capture image(s) */}
```
Comment (Wajeed-msft) [inline samples/tab-device-permissions/nodejs/src/components/Tab.tsx:18]: Can we render these in a grid like view? Current view looks fine only on mobile.

**Review by Wajeed-msft — CHANGES_REQUESTED** [review]

File: samples/tab-device-permissions/nodejs/README.md (hunk 12 above)
Comment (v-kalc) [reply samples/tab-device-permissions/nodejs/README.md]: resolved the conflits

**Review by Wajeed-msft — APPROVED** [review]
